sdk/core/azure-core-management/src/main/java/com/azure/core/management/provider/DelayProvider.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.management.provider;
+
+import java.time.Duration;
+
+/**
+ * The DelayProvider to help thread sleep.
+ */
+public interface DelayProvider {
+
+    /**
+     * Thread sleep.
+     *
+     * @param milliseconds time for sleep
+     */
+    void sleep(int milliseconds);
```

Comment:
Simliarly, this is only `Thread.sleep(getDelayDuration(Duration.ofMillis(milliseconds)).toMillis())`, and can be in a helper method elsewhere.